src/GoTrueClient.ts

```diff
@@ -807,31 +790,15 @@ export default class GoTrueClient {
   /**
    * Acquires a global lock based on the storage key.
    */
-  private async _acquireLock<R>(acquireTimeout: number, fn: () => Promise<R>): Promise<R> {
+  async _acquireLock<R>(acquireTimeout: number, fn: () => Promise<R>): Promise<R> {
     this._debug('#_acquireLock', 'begin', acquireTimeout)
 
     try {
-      if (!(await stackGuardsSupported())) {
-        this._debug(
-          '#_acquireLock',
-          'Stack guards not supported, so exclusive locking is not performed as it can lead to deadlocks if the lock is attempted to be recursively acquired (as the recursion cannot be detected).'
-        )
-
-        return await fn()
-      }
-
-      if (isInStackGuard('_acquireLock')) {
-        this._debug('#_acquireLock', 'recursive call')
-        return await fn()
-      }
-
       return await this.lock(`lock:${this.storageKey}`, acquireTimeout, async () => {
```

Comment:
just to clarify, the default implementation of the lock doesn't do anything with the `name` and `acquireTimeout` arguments right?